### ISSUE TITLE
Problem with jetson orin nano and version ultralytics >= 8.3.5

### ISSUE THREAD

**Issue by VladimirMbf** [ultralytics/ultralytics#18633]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Predict

### Bug

Hello!

Could you help me please, I have jetson orin nano with jetpack 6.1 rev 1 installed, I have a problem with using ultralytics with version higher 8.3.4

The problem is with a inference time on object counting, I tried to use different models from yolov8(from n to m types of models .pt, .engine- fp32, fp16 and int8, onnx) to yolo11(same as yolov8) any of these models shows 27-30ms inference time all the time and I didn't see gpu heavy load

ultralytics <= 8.3.4 works well (.engine works as usual with different models from fp32 to int8) but tracking working much better in higher versions of ultralytics

Can you please tell me why this can happen and how to make normal working with latest versions of ultralytics?

### Environment

Ultralytics 8.3.4 🚀 Python-3.10.12 torch-2.5.0a0+872d972e41.nv24.08 CUDA:0 (Orin, 3602MiB)
Setup complete ✅ (6 CPUs, 3.5 GB RAM, 58.8/232.2 GB disk)

OS                  Linux-5.15.148-tegra-aarch64-with-glibc2.35
Environment         Linux
Python              3.10.12
Install             pip
RAM                 3.52 GB
CPU                 ARMv8 Processor rev 1 (v8l)
CUDA                12.6

numpy               ✅ 1.23.5>=1.23.0
matplotlib          ✅ 3.10.0>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 11.1.0>=7.1.2
pyyaml              ✅ 5.4.1>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.15.0>=1.4.1
torch               ✅ 2.5.0a0+872d972e41.nv24.8>=1.8.0
torchvision         ✅ 0.20.0a0+afc54f7>=0.9.0
tqdm                ✅ 4.67.1>=4.64.0
psutil              ✅ 5.9.8
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 1.3.5>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.13>=2.0.0
torch               ✅ 2.5.0a0+872d972e41.nv24.8!=2.4.0,>=1.8.0; sys_platform == "win32"

### Minimal Reproducible Example

... 

    search_class = 1

    object_class = {0: "person", 1: "bicycle"}

    count_object = object_class.get(search_class)

    count_class = [search_class]

    ver = int(ultralytics.__version__.replace('.', ''))
    print(f'Version: {ver}')

    if ver <= 834:
        counter = solutions.ObjectCounter(
            reg_pts=line_points,
            # classes=[search_class],
            # verbose=False
            # view_img=False,
            names=model.names,  # {0: 'person', 1: 'bicycle'},
            draw_tracks=True,
            line_thickness=2,
        )
    else:
        counter = solutions.ObjectCounter(
            region=line_points,
            # reg_pts=line_points,
            classes=[search_class],
            # verbose=False
            # draw_tracks=True,
            # line_thickness=2,
        )

                    frame = cv2.cvtColor(frame, cv2.COLOR_BGRA2BGR)

                    if ver <= 834:

                        tracks = model.track(frame, persist=True,                                            
                                             classes=count_class,
                                             # half=True,
                                             tracker="bytetrack.yaml",
                                             stream=True,
                                             # visualize=True,
                                             # augment=True,
                                             # retina_masks=True,
                                             # agnostic_nms=True,
                                             # stream=True,
                                             # imgsz=640,
                                             # verbose=False,
                                             batch=4
                                             )

                        tracks = next(tracks)
                        tracks = [tracks]
                        frame = counter.start_counting(frame, tracks)

                    else:
                        model.track(frame, persist=True,
                                    classes=count_class,
                                    device='cuda:0',
                                    # half=True,
                                    # tracker="bytetrack.yaml",
                                    stream=True,
                                    # visualize=True,
                                    # augment=True,
                                    # retina_masks=True,
                                    # agnostic_nms=True,
                                    # stream=True,
                                    # imgsz=640,
                                    # verbose=False,
                                    batch=4
                                    )

                        frame = counter.count(frame)

Inference time with ultralytics 8.3.4, model yolo11m.engine int8 from mp4 file

0: 640x640 4 bicycles, 23.6ms

0: 640x640 4 bicycles, 24.6ms

0: 640x640 4 bicycles, 24.4ms

0: 640x640 4 bicycles, 24.0ms

0: 640x640 4 bicycles, 24.1ms

0: 640x640 4 bicycles, 24.1ms

0: 640x640 4 bicycles, 23.9ms

Inference time with ultralytics 8.3.59, model yolo11m.engine int8 from the same mp4 file

0: 384x640 11 bicycles, 169.0ms
Speed: 7.7ms preprocess, 169.0ms inference, 172.7ms postprocess per image at shape (1, 3, 384, 640)

0: 384x640 11 bicycles, 36.1ms
Speed: 7.7ms preprocess, 36.1ms inference, 5.3ms postprocess per image at shape (1, 3, 384, 640)

0: 384x640 12 bicycles, 29.2ms
Speed: 6.7ms preprocess, 29.2ms inference, 3.9ms postprocess per image at shape (1, 3, 384, 640)

0: 384x640 11 bicycles, 25.9ms
Speed: 6.6ms preprocess, 25.9ms inference, 3.6ms postprocess per image at shape (1, 3, 384, 640)

0: 384x640 13 bicycles, 30.7ms
Speed: 6.6ms preprocess, 30.7ms inference, 4.9ms postprocess per image at shape (1, 3, 384, 640)

0: 384x640 13 bicycles, 28.6ms
Speed: 6.5ms preprocess, 28.6ms inference, 3.9ms postprocess per image at shape (1, 3, 384, 640)

but if I try to use any different models as I marked above I see the same inference time

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @VladimirMbf, thank you for your detailed report and your interest in Ultralytics 🚀! We greatly appreciate the effort you've put into outlining the issue. An Ultralytics engineer will assist you soon to further investigate this.

In the meantime, here are some initial suggestions that may help:

If this is a 🐛 Bug Report:
- Please ensure you've provided a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum_reproducible_example/). Your detailed script is super helpful, but having the smallest piece of code that consistently replicates the issue can speed up our debugging process.

For questions regarding optimization or performance:
- Please share specific logs or benchmarks (if possible) to help us understand the discrepancy between versions. Detailed insights into GPU usage and inference logs are especially beneficial.

## Suggestions to Try

### Upgrade
To ensure you're running the latest fixes and improvements, try upgrading to the latest `ultralytics` package in a [**Python>=3.8**](https://www.python.org/) environment with the necessary dependencies like [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

It's also worth verifying dependency compatibility, especially with the hardware-specific libraries on your Jetson Orin Nano.

### Environments for Consistency
YOLO is verified in a variety of environments to ensure consistency. While Jetson platforms may have some unique quirks, consider trying these setups for cross-verification:
- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Docker Image** with preconfigured dependencies: See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Community and Documentation

- Our [Docs](https://docs.ultralytics.com) are constantly updated with new features and usage guides. For training-related insights, check the [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) 📚.
- For real-time assistance, join our [Discord](https://discord.com/invite/ultralytics) community 🎧 or post on [Discourse](https://community.ultralytics.com) for discussions.

## CI Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all core functionalities have passed our comprehensive nightly tests for macOS, Windows, and Ubuntu. However, differences in hardware like Jetson Orin Nano may still lead to unique behaviors that we'll investigate.

We’ll keep this issue open until we have a resolution or further instructions for you. Thank you for your patience and for being a part of our Ultralytics community! 😊

**Comment by Y-T-G** [comment]
Did you try latest version?

**Comment by VladimirMbf** [comment]
Hello @Y-T-G 

Yes, you're right I tried latest version 8.3.59 too. I have same problem from 8.3.5 till 8.3.59 (latest)

**Comment by Y-T-G** [comment]
Are you running inside docker?

**Comment by VladimirMbf** [comment]
No, because docker is making more delays I'm not using any containers in docker

**Comment by Y-T-G** [comment]
Can you post the minimum reproducible code? Only the code that causes the issue. Does it occur only if you use the counter? Otherwise, remove the counter code. Only keep the code that causes the issue.

**Comment by VladimirMbf** [comment]
Hm, you're right the problem is with counting, because prediction works fast

# model.track(frame, persist=True,
                        #             classes=count_class,
                        #             device='cuda:0',
                        #             # half=True,
                        #             tracker="bytetrack.yaml",
                        #             stream=True,
                        #             # visualize=True,
                        #             # augment=True,
                        #             # retina_masks=True,
                        #             # agnostic_nms=True,
                        #             # stream=True,
                        #             # imgsz=640,
                        #             verbose=True,
                        #             batch=8
                        #             )

                        # frame = counter.count(frame)


results = model.predict(frame,
                                                classes=count_class,
                                                device='cuda:0')[0]

frame = results[0].plot()


0: 384x640 9 bicycles, 15.5ms
Speed: 6.4ms preprocess, 15.5ms inference, 4.1ms postprocess per image at shape (1, 3, 384, 640)

0: 384x640 7 bicycles, 15.4ms
Speed: 6.5ms preprocess, 15.4ms inference, 4.2ms postprocess per image at shape (1, 3, 384, 640)

0: 384x640 9 bicycles, 15.4ms
Speed: 6.5ms preprocess, 15.4ms inference, 4.2ms postprocess per image at shape (1, 3, 384, 640)


Tell me please, how we can fix counting? :)

**Comment by Y-T-G** [comment]
@RizwanMunawar Can you check this?

**Comment by VladimirMbf** [comment]
@Y-T-G, ok no problem I’ll send you gpu utilisation on both tracking and counting but after one, two hours please

And thank you for the problem explanation

Can you tell me please, is there some way to make some temporary fix until solutions v1.1 will come?

**Comment by Y-T-G** [comment]
Is the issue with counter or tracking?

**Comment by VladimirMbf** [comment]
Sorry, right now my notebook not near me, but last time I saw the problem only with counting because tracking was uncommented and everything worked good (but I will check this again and write you). Now the problem as I understand with counting only tracking is ok

**Comment by VladimirMbf** [comment]
I checked tracking and counting

Tracking is working with same inference time as prediction ~15ms, gpu load on start 87% on process from 5% to 60%. Tracking is working well

Counting, I found another thing
1. counting with stream=True, tracking inference time 27-30ms gpu load on 34% on process from 2% to 60% (almost same)
2. counting with stream=False, tracking inference one frame 15ms second frame 25-27ms and all frames 15ms, 27ms, 15ms, 25ms, 15ms, 26ms, ...

The problem in counting only

**Comment by VladimirMbf** [comment]
@Y-T-G I would really appreciate it if you could provide a temporary fix with a counting please, if this's possible

**Comment by RizwanMunawar** [comment]
@VladimirMbf Hi, please note you can't get the same speed with counting, as counting also involves the additional logic to count objects, and it will take more time in comparison with simple object tracking. 

However, please let me know If I can help you with something else. Thanks

**Comment by VladimirMbf** [comment]
@RizwanMunawar, can you tell me please where in 8.3.59 I can disable draw_tracks?

**Comment by VladimirMbf** [comment]
Thank you so much @RizwanMunawar @Y-T-G for your help and support!

I don't know why and how (need to have some research) but with AI I made counting code with opencv (it using tracking from ultralytics) and it's working with same speed ~15ms (instead 24 or 30) per frame (but without object tracking points). First test video shows same results as in ultralytics 8.3.4. I'll continue testing
